### PR TITLE
Extract loadTargetFromFile into it's own module

### PR DESCRIPTION
**Pull request by CameronDiver**
This is the start of some work on refactoring the device-state module, starting with removing some non-related code into other places, and then refactoring the class itself.

Also define some more types for interacting with target states, and upgrade typescript.

### CODE REVIEW

File: src/application-manager.d.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { EventTracker } from './event-tracker';
 import { Logger } from './logger';
 import { DeviceApplicationState } from './types/state';
 
-import Images from './compose/images';
+import Images, { Images } from './compose/images';
```
Comment (zvin) [inline src/application-manager.d.ts]: Oo

Comment (CameronDiver) [reply src/application-manager.d.ts]: uhh, supposed to be `Image` (this needs refactored too). It's weird that this even builds though, much less passes the tests...

File: src/device-state/preload.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+): Promise<void> {
+	log.info('Attempting to load any preloaded applications');
+	if (!appsPath) {
+		appsPath = constants.appsJsonPath;
+	}
+
+	try {
+		const content = await fs.readFile(appsPath, 'utf8');
+
+		// It's either a tgarget state
```
Comment (zvin) [inline src/device-state/preload.ts]: tgarget -> target

Also:

There are two types of people in this world:
1. Those who can extrapolate from incomplete data.

File: src/device-state/preload.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		const localState = { local: { name: '', ...preloadState } };
+
+		await deviceState.setTarget(localState);
+
+		log.success('Preloading complete');
+		if (stateFromFile.pinDevice) {
+			// Multi-app warning!
+			// The following will need to be changed once running
+			// multiple applications is possible
+			log.debug('Device will be pinned');
```
Comment (zvin) [inline src/device-state/preload.ts]: NIT: the debug message could be inside the if below

File: src/lib/migration.ts [hunk 4]
```diff
@@ -67,3 +69,18 @@ export function singleToMulticontainerApp(app: Dictionary<any>): Application {
 	};
 	return newApp;
 }
+
+export async function convertLegacyAppsJson(
```
Comment (zvin) [inline src/lib/migration.ts]: This function does not need to be `async`.

**Comment by CameronDiver** [comment]
Tested on an rpi3 and works great.

**Pull request merged**